shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetServiceImpl.java

```diff
@@ -253,8 +257,34 @@ public List<Dataset> findByStudyId(Long studyId) {
 	}
 
 	@Override
-	public Long getExpressionSizeByStudyId(Long studyId) {
-		return repository.getExpressionSizeByStudyId(studyId);
+	public List<VolumeByFormatDTO> getVolumeByFormat(Long studyId) {
+		List<Object[]> results = repository.findExpressionSizesByStudyIdGroupByFormat(studyId);
+		List<VolumeByFormatDTO> sizesByFormat = new ArrayList<>();
+
+		for(Object[] result : results){
```

Comment:
I was wondering, if there is no data, does it still returns something usable here ?